cocos/animation/marionette/pose-graph/pose-nodes/ik/two-bone-ik-solver.ts

```diff
@@ -0,0 +1,260 @@
+import { EDITOR } from 'internal:constants';
+import { ccclass, editable, serializable, type, visible } from '../../../../../core/data/decorators';
+import { CLASS_NAME_PREFIX_ANIM } from '../../../../define';
+import { PoseTransformSpaceRequirement } from '../../pose-node';
+import { assertIsTrue, ccenum, Vec3 } from '../../../../../core';
+import { TransformHandle } from '../../../../core/animation-handle';
+import { input } from '../../decorator/input';
+import { poseGraphNodeCategory } from '../../decorator/node';
+import { Pose } from '../../../../core/pose';
+import { PoseNodeModifyPoseBase, TransformModificationQueue } from '../modify-pose-base';
+import { solveTwoBoneIK } from './solve-two-bone-ik';
+import { Transform } from '../../../../core/transform';
+import { AnimationGraphBindingContext, AnimationGraphEvaluationContext } from '../../../animation-graph-context';
+import { PoseGraphType } from '../../foundation/type-system';
+import { TransformSpace } from '../transform-space';
+import { POSE_GRAPH_NODE_MENU_PREFIX_IK } from './menu';
+
+const cacheRootTransform = new Transform();
+const cacheMiddleTransform = new Transform();
+const cacheEndEffectorTransform = new Transform();
+const cacheEndEffectorTargetPosition = new Vec3();
+const cachePoleTargetPosition = new Vec3();
+const cacheTransform_evaluateTarget = new Transform();
+
+enum TargetSpecificationType {
+    /**
+     * Targets nothing.
+     */
+    NONE,
+
+    /**
+     * Targets the specified vector value.
+     */
+    VALUE,
+
+    /**
+     * Targets the specified bone.
+     */
+    BONE,
+}
+
+ccenum(TargetSpecificationType);
+
+@ccclass(`${CLASS_NAME_PREFIX_ANIM}PoseNodeTwoBoneIKSolver.TargetSpecification`)
+class TargetSpecification {
+    constructor (type?: TargetSpecificationType) {
+        this.type = type ?? TargetSpecificationType.NONE;
+    }
+
+    @serializable
+    @editable
+    @type(TargetSpecificationType)
+    public type = TargetSpecificationType.VALUE;
+
+    @serializable
+    @editable
+    @visible(function (this: TargetSpecification) { return this.type === TargetSpecificationType.VALUE; })
```

Comment:
ESlint unnamed function.

Reply:
Solved in f4a158975ca9629e1bd7969108f186b14c3de1c4 .